service/src/main/java/edu/tamu/catalog/exception/RestExceptionHandler.java

```diff
@@ -15,19 +15,36 @@
 @ControllerAdvice
 public class RestExceptionHandler extends ResponseEntityExceptionHandler {
 
+    private static final String RENEWAL_DID_NOT_CHANGE_THE_DUE_DATE = "Renewal did not change the due date";
+
     private final Logger logger = LoggerFactory.getLogger(this.getClass());
 
+
+    @ExceptionHandler(RenewFailureException.class)
+    public ResponseEntity<String> renewError(RenewFailureException e, WebRequest request) {
+        logger.warn(e.getMessage());
+
+        if (logger.isDebugEnabled()) {
+            e.printStackTrace();
+        }
+
+        return ResponseEntity.status(HttpStatus.UNPROCESSABLE_ENTITY)
+            .contentType(MediaType.TEXT_PLAIN)
+            .body(RENEWAL_DID_NOT_CHANGE_THE_DUE_DATE);
+    }
+
     @ExceptionHandler(HttpClientErrorException.class)
     public ResponseEntity<String> clientError(HttpClientErrorException e, WebRequest request) {
         logger.warn(e.getMessage());
 
         if (logger.isDebugEnabled()) {
+            logger.warn(e.getResponseBodyAsString());
             e.printStackTrace();
         }
 
         return ResponseEntity.status(e.getRawStatusCode())
-            .contentType(MediaType.TEXT_PLAIN)
-            .body(e.getMessage());
+            .contentType(MediaType.APPLICATION_JSON)
+            .body(e.getResponseBodyAsString());
```

Comment:
Does `e.getResponseBodyAsString()` return a json string?
Or Does ResponseEntity return a string as a json?

If not, I would think that text/plain is what is actually being returned here and not json.

I thought the exceptions returned are returned as text/plain.

Reply:
HTTP Response Body is always a string. In this case, it is a JSON string. In these cases, we need to return the response that FOLIO is returning to get the information required. I think we need to further determine what exactly we want our API status and response types.